lib/collection/property.js

```diff
@@ -170,6 +170,17 @@ _.extend(Property, /** @lends Property */ {
      * @returns {String}
      */
     replaceSubstitutions: function (str, variables) {
+        var resolvedStr,
+            replaced = true,
+            loopCounter = 0,
+            onReplace = function (match, token) {
```

Comment:
The definition of replace function should go down after current line 190 since the nature of `variables` is modified at that line and also there is a validation function bailout at line 185 and there is no point doing a function definition if the validations fail.